readthedocs/projects/tasks/builds.py

```diff
@@ -514,28 +512,38 @@ def on_failure(self, exc, task_id, args, kwargs, einfo):
         self.data.build['success'] = False
 
     def on_success(self, retval, task_id, args, kwargs):
-        html = self.data.outcomes['html']
-        search = self.data.outcomes['search']
-        localmedia = self.data.outcomes['localmedia']
-        pdf = self.data.outcomes['pdf']
-        epub = self.data.outcomes['epub']
-
         time_before_store_build_artifacts = timezone.now()
         # Store build artifacts to storage (local or cloud storage)
-        self.store_build_artifacts(
-            html=html,
-            search=search,
-            localmedia=localmedia,
-            pdf=pdf,
-            epub=epub,
-        )
+        #
+        # TODO: move ``store_build_artifacts`` inside ``execute`` so we can
+        # handle exceptions properly on ``on_failure``
+        self.store_build_artifacts()
         log.info(
             "Store build artifacts finished.",
             time=(timezone.now() - time_before_store_build_artifacts).seconds,
         )
 
-        # NOTE: we are updating the db version instance *only* when
         # HTML are built successfully.
```

Comment:
Seems we deleted half this comment. Is it not true anymore? 

Ah. I see it got missed in a comment refactor. This line needs to be moved down below :)

Reply:
I got confused by this comment. I'm not sure if there is something I have to do here or not 😅

Reply:
Yes, the comment was 2 lines, and only 1 line was moved a few lines down. The comment is basically saying we only hit the API when `html` passes, which is medium-weird, but we don't want to change with this refactor.

Reply:
> which is medium-weird, but we don't want to change with this refactor

Yes, this is _wrong_, in my opinion. I noticed this when I did the Celery refactor. This is something I want to change soon, if possible.